### PR TITLE
Separate out the index generating function in the ProportionalSplit class.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: ethicml/preprocessing/train_test_split.py [hunk 1]
```diff
@@ -160,6 +160,46 @@ def __call__(
         return train_test_split(data, self.train_percentage, random_seed) + (split_info,)
 
 
+def generate_proportional_split_indexes(
+    data: DataTuple, train_percentage, random_seed: int = 42
+) -> Tuple["np.ndarray[np.int64]", "np.ndarray[np.int64]"]:
```
Comment (olliethomas) [inline ethicml/preprocessing/train_test_split.py]: add `from __future__ import annotations` at the top of the file and you won't need the quotes

Comment (olliethomas) [reply ethicml/preprocessing/train_test_split.py]: nevermind. Apparently we still support 3.6

File: ethicml/preprocessing/train_test_split.py [hunk 2]
```diff
@@ -160,6 +160,46 @@ def __call__(
...
+    # local random state that won't affect the global state
+    random = RandomState(seed=random_seed)
+
+    s_col = data.s.columns[0]
+    y_col = data.y.columns[0]
+
+    s_vals: List[int] = list(map(int, data.s[s_col].unique()))
+    y_vals: List[int] = list(map(int, data.y[y_col].unique()))
+
+    train_empty: List[int] = []  # this is necessary because mypy is silly
```
Comment (olliethomas) [inline ethicml/preprocessing/train_test_split.py]:
```suggestion
    train_empty: List[int] = []
```

File: ethicml/preprocessing/train_test_split.py [hunk 3]
```diff
@@ -160,6 +160,48 @@ def __call__(
         return train_test_split(data, self.train_percentage, random_seed) + (split_info,)
 
 
+def generate_proportional_split_indexes(
+    data: DataTuple, train_percentage, random_seed: int = 42
```
Comment (thomkeh) [inline ethicml/preprocessing/train_test_split.py]:
```suggestion
    data: DataTuple, train_percentage: float, random_seed: int = 42
```

**Pull request merged**